### PR TITLE
Improve ListenBrainz error handling and simplify playlist handling

### LINKED ISSUE #5459: Listenbrainz broken - AttributeError
Listenbrainz is currently broken. I get this error:

```bash
Traceback (most recent call last):
  File "/home/arsaboo/.local/bin/beet", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/arsaboo/.local/lib/python3.12/site-packages/beets/ui/__init__.py", line 1862, in main
    _raw_main(args)
  File "/home/arsaboo/.local/lib/python3.12/site-packages/beets/ui/__init__.py", line 1849, in _raw_main
    subcommand.func(lib, suboptions, subargs)
  File "/home/arsaboo/.local/lib/python3.12/site-packages/beetsplug/plexsync.py", line 342, in func_playlist_import
    self._plex_import_playlist(opts.playlist, opts.url, opts.listenbrainz)
  File "/home/arsaboo/.local/lib/python3.12/site-packages/beetsplug/plexsync.py", line 947, in _plex_import_playlist
    weekly_jams = lb.get_weekly_jams()
                  ^^^^^^^^^^^^^^^^^^^^
  File "/home/arsaboo/.local/lib/python3.12/site-packages/beetsplug/listenbrainz.py", line 258, in get_weekly_jams
    return self.get_weekly_playlist(1)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/arsaboo/.local/lib/python3.12/site-packages/beetsplug/listenbrainz.py", line 250, in get_weekly_playlist
    return self.get_tracks_from_playlist(playlist)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/arsaboo/.local/lib/python3.12/site-packages/beetsplug/listenbrainz.py", line 205, in get_tracks_from_playlist
    "identifier": track.get("identifier").split("/")[-1],
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'split'
```

I will submit a PR soon to fix this.

### PR DESCRIPTION
**Pull request by arsaboo**
## Description

Fixes #5459

## To Do
- [X] Changelog. (Add an entry to `docs/changelog.rst` to the bottom of one of the lists near the top of the document.)
- [ ] Tests. (Very much encouraged but not strictly required.)

### CODE REVIEW

File: beetsplug/listenbrainz.py [hunk 1]
```diff
@@ -199,17 +181,21 @@ def get_tracks_from_playlist(self, playlist):
...
+                identifier = identifier[0]
+            artist = track.get(
+                "creator", "Unknown artist"
+            )  # Set a default value if 'creator' key is not present
+            track_dict = {
+                "artist": artist,
+                "identifier": identifier.split("/")[-1],
+                "title": track.get("title"),
+            }
+            tracks.append(track_dict)
```
Comment (snejus) [inline beetsplug/listenbrainz.py:195]: I find the original implementation slightly cleaner to read. Consider something like below
```suggestion
            identifier = track.get("identifier")
            if isinstance(identifier, list):
                identifier = identifier[0]

            tracks.append(
                {
                    "artist": track.get("creator", "Unknown artist"),
                    "identifier": identifier.split("/")[-1],
                    "title": track.get("title"),
                }
            )
```

**Review by snejus — CHANGES_REQUESTED** [review]
I commented under the change that is required to fix #5459.

Could you by any chance submit the rest of the changes in a separate PR as they seem to be a more general refactor?

Also, I suspect you are using black to do auto formatting - note that the max line length is configured as **80** across the project.

This should have been caught by CI, but now I see that `ruff` only  shouts for line lengths longer than 88.

**Comment by arsaboo** [comment]
@snejus Made the changes. It looks like the check is failing for unrelated reasons.

**Comment by arsaboo** [comment]
Tests are not my strong suit unfortunately...

The current version is broken anyways, so I don't think we can be worse off. I'm here in case there are any issues that come up. I use this plugin regularly, so trust me I'll be the first to identify an issue 😜

File: beetsplug/listenbrainz.py [hunk 2]
```diff
@@ -159,35 +156,32 @@ def get_listenbrainz_playlists(self):
...
-                    )
+                listenbrainz_playlists.append(
+                    {"type": playlist_type, "date": date, "identifier": id}
+                )
+        listenbrainz_playlists = sorted(
+            listenbrainz_playlists, key=lambda x: x["type"]
+        )
+        listenbrainz_playlists = sorted(
+            listenbrainz_playlists, key=lambda x: x["date"], reverse=True
+        )
```
Comment (bal-e) [inline beetsplug/listenbrainz.py:180]: Is the intent here to sort everything by date, and between playlists of the same date, to sort by type?  Setting `key=lambda x: (x["date"], x["type"])` would be more explicit and more efficient.

**Review by bal-e — COMMENTED** [review]
This is a nice and simple little change, thanks @arsaboo!

File: beetsplug/listenbrainz.py (hunk 2 above)
Comment (bal-e) [reply beetsplug/listenbrainz.py:180]: Oh, never mind, I see that sorting one key in reverse might be a bit difficult.

**Pull request merged**